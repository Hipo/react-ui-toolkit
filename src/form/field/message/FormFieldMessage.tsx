import "./_form-field-message.scss";

import React from "react";
import classNames from "classnames";

export interface FormFieldMessageProps {
  type: "error" | "warning" | "helper";
  customClassName?: string;
  message?: string;
  testid?: string;
}

function FormFieldMessage({
  customClassName,
  type,
  message,
  testid
}: FormFieldMessageProps) {
  return (
    <p
      data-testid={testid}
      className={classNames(
        "form-field-message",
        customClassName,
        `form-field-message--is-${type}`
      )}
    >
      {message}
    </p>
  );
}

export default FormFieldMessage;
