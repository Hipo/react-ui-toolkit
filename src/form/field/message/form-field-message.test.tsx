import React from "react";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom";
import {create} from "react-test-renderer";

import {testA11y} from "../../../core/utils/test/testUtils";
import FormFieldMessage, {FormFieldMessageProps} from "./FormFieldMessage";

describe("<FormFieldMessage />", () => {
  afterEach(cleanup);

  const defaultFormFieldMessageProps: FormFieldMessageProps = {
    testid: "form-field-message",
    type: "error"
  };

  it("should render correctly", () => {
    render(<FormFieldMessage {...defaultFormFieldMessageProps} />);
  });

  it("should match snapshot", () => {
    const tree = create(<FormFieldMessage {...defaultFormFieldMessageProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should pass a11y test", async () => {
    const {container} = render(<FormFieldMessage {...defaultFormFieldMessageProps} />);

    await testA11y(container);
  });

  it("should render message correctly", () => {
    const {getByTestId} = render(
      <FormFieldMessage message={"Test Message"} {...defaultFormFieldMessageProps} />
    );

    expect(getByTestId(defaultFormFieldMessageProps.testid!)).toHaveTextContent(
      "Test Message"
    );
  });

  it("should have proper modifier class name for the type prop provided", () => {
    const {rerender, getByTestId} = render(
      <FormFieldMessage {...defaultFormFieldMessageProps} />
    );

    const formFieldMessageTypes: typeof defaultFormFieldMessageProps.type[] = [
      "error",
      "warning",
      "helper"
    ];

    formFieldMessageTypes.forEach((type) => {
      rerender(<FormFieldMessage {...defaultFormFieldMessageProps} type={type} />);

      expect(getByTestId(defaultFormFieldMessageProps.testid!)).toHaveClass(
        `form-field-message--is-${type}`
      );
    });
  });
});
