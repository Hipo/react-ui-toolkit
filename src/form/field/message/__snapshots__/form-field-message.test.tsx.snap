// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFieldMessage /> should match snapshot 1`] = `
<p
  className="form-field-message form-field-message--is-error"
  data-testid="form-field-message"
>
  Form Field Message Test
</p>
`;
